libs/designer-ui/src/lib/settings/settingsection/settingTokenField.tsx

```diff
@@ -315,3 +328,38 @@ const TokenField = ({
       );
   }
 };
+
+export type CustomTokenFieldProps = TokenFieldProps & { editorOptions: ICustomEditorOptions };
+
+export const CustomTokenField = (props: CustomTokenFieldProps) => {
+  invariant(isCustomEditor(props), 'only custom editor can be rendered by CustomTokenField');
+  const { EditorComponent, editor, editorOptions = {} }: ICustomEditorOptions = props.editorOptions;
+
+  const renderDefaultEditor = useRenderDefaultEditor(props);
```

Comment:
instead of using the default editor, and omitting props, we can just import the UI editor component in our custom render. This logic shouldn't be within the SettingTokenField file